### ISSUE TITLE
Workaround for dnsmasq problem

### ISSUE THREAD

**Issue by yichya** [yichya/luci-app-xray#425]
https://github.com/openwrt/openwrt/pull/14975 involves a change to dnsmasq configuration path (`/tmp/dnsmasq.d` -> `/tmp/dnsmasq.cfg01411c.d`) which breaks DNS configurations for luci-app-xray. Add this to Custom Configurations as a workaround:

```js
return function(config) {
    const fs = require("fs");
    const uci = require("uci");
    const cursor = uci.cursor();
    const first = cursor.get_first("dhcp", "dnsmasq");

    fs.mkdir("/tmp/dnsmasq.d");
    fs.symlink("/tmp/dnsmasq.d/xray.conf", `/tmp/dnsmasq.${first}.d/xray.conf`);   
}
```

**Comment by yichya** [comment]
Caution: if you're using OpenWrt with ujail enabled, the workaround above won't work and may even make dnsmasq unable to start at all. Manually edit `/etc/init.d/dnsmasq` to add `/tmp/dnsmasq.d` to jail mount before applying workaround above: 

```diff
diff --git a/package/network/services/dnsmasq/files/dnsmasq.init b/package/network/services/dnsmasq/files/dnsmasq.init
index b864ea9069..7a535f576d 100755
--- a/package/network/services/dnsmasq/files/dnsmasq.init
+++ b/package/network/services/dnsmasq/files/dnsmasq.init
@@ -1253,7 +1253,7 @@ dnsmasq_start()
 
        procd_add_jail dnsmasq ubus log
        procd_add_jail_mount $CONFIGFILE $DHCPBOGUSHOSTNAMEFILE $DHCPSCRIPT $DHCPSCRIPT_DEPENDS
-       procd_add_jail_mount $EXTRA_MOUNT $RFC6761FILE $TRUSTANCHORSFILE
+       procd_add_jail_mount $EXTRA_MOUNT $RFC6761FILE $TRUSTANCHORSFILE /tmp/dnsmasq.d
        procd_add_jail_mount $dnsmasqconffile $dnsmasqconfdir $resolvdir $user_dhcpscript
        procd_add_jail_mount /etc/passwd /etc/group /etc/TZ /etc/hosts /etc/ethers
        procd_add_jail_mount_rw /var/run/dnsmasq/ $leasefile
```

A better solution for this problem will be included in version 3.5 of this project (which is not planned before, but seems necessary now).

**Comment by yichya** [comment]
https://github.com/yichya/luci-app-xray/pull/430 includes an option to set dnsmasq integration mode in preview app.

<img width="978" alt="image" src="https://github.com/user-attachments/assets/da1c54ce-5694-4de4-b46a-e71979191dbc">

If luci-app-xray works correctly, don't change this option. Only try "Per Instance" mode if:

* You're encountering problems related to https://github.com/openwrt/openwrt/pull/14975
    * Select "Default instance"
* You're using other DNS services like AdGuard Home and it somewhat conflicts with luci-app-xray
    * Select none to disable any dnsmasq integrations
    * Make sure your DNS services are working correctly
    * Use 127.0.0.1:5300 (this is still provided by luci-app-xray) as an upstream server for your DNS services
    * You may not be able to associate domain names in manual transparent proxy under this circumstance (because of being unable to exempt Rebind Protection). Use some public blackholes (for example [NYIIX RTBH](https://www.nyiix.net/technical/rtbh/)) to avoid that.
* You're really setting up multiple dnsmasq instances and only some of them needs luci-app-xray
